### ISSUE TITLE
Timeout to ftp.halifax.rwth-aachen.de

### ISSUE THREAD

**Issue by KweezyCode** [jenkins-infra/helpdesk#4128]
### Service(s)

get.jenkins.io

### Summary

Caused: java.io.IOException: Failed to download from https://updates.jenkins.io/download/plugins/workflow-api/1316.v33eb_726c50b_a_/workflow-api.hpi (redirected to: https://ftp.halifax.rwth-aachen.de/jenkins/plugins/workflow-api/1316.v33eb_726c50b_a_/workflow-api.hpi)

### Reproduction steps

location: Russia (Vimpelcom)

**Comment by KweezyCode** [comment]
> Hello @KweezyCode,
> 
> You're probably in the same situation as #4096 & #4105, I suggest you to send a mail to [ftp@halifax.rwth-aachen.de](mailto:ftp@halifax.rwth-aachen.de) explaining your issue.

how could i change mirror? it would be faster anyway

**Comment by C-Otto** [comment]
I doubt it. I would have responded by now.

**Comment by dduportal** [comment]
> > Hello @KweezyCode,
> > You're probably in the same situation as #4096 & #4105, I suggest you to send a mail to [ftp@halifax.rwth-aachen.de](mailto:ftp@halifax.rwth-aachen.de) explaining your issue.
> 
> how could i change mirror? it would be faster anyway

Hi!

- The mirror redirector system does not allow to specify a custom mirror as far as we know. It's using GeoIP location.
- As soon as https://github.com/jenkins-infra/helpdesk/issues/3976 is finished, there should be a closer mirror hosted in Romania. Of course, iy you know a company or organization close to your location who would be able to host a mirror (need webserver, a rsync or FTP system, and ~750 Gb of disk), we would be happy to add it!

I'll close the issue as there are no actionnable for the Jenkins infra team:
- You have to contact ftp@halifax.rwth-aachen.de to solve your problem

**Comment by Stikus** [comment]
@dduportal Correct me if I'm wrong, but if its our ISP problem and we cannot switch ISP we should use any method to change our IP so GeoIP would pick another mirror for us?

I've contacted @C-Otto (thx for fast answer again) - our IP is not blocked by `ftp@halifax.rwth-aachen.de`

**Comment by C-Otto** [comment]
If it's a problem caused by your ISP, you should ask them to fix it (and postpone payments until they do).

**Comment by mikhirev** [comment]
Hi!

This problem is not caused by ISP. The ftp.halifax.rwth-aachen.de server does not respond to any connect attempts from Russian IP addresses. Can you change your GeoIP settings to exclude this server for requests from Russia?

**Comment by Stikus** [comment]
We are trying to connect our ISP now, but our network engineers confirming, that all Russian ISP have this problem and they didn't find legal reason for that (any government act or so).

**Comment by KweezyCode** [comment]
just tried with different providers. Yea, looks like all russian ips banned

**Comment by C-Otto** [comment]
> just tried with different providers. Yea, looks like all russian ips banned

It's the reverse. We don't ban Russian IPs. Instead, Russia (the state/government?) blocks our IP address.

**Comment by KweezyCode** [comment]
> > just tried with different providers. Yea, looks like all russian ips banned
> 
> It's the reverse. We don't ban Russian IPs. Instead, Russia (the state/government?) blocks our IP address.

it is not, ip or/and domain are not in russian blocklists

**Comment by C-Otto** [comment]
Well, then it's just a very weird coincidence that the reports from Russian users started to come in shortly after we added a Tor relay (which can be used for free speech, anti-censorship, ... - concepts I don't think are at home in Russia).

**Comment by Stikus** [comment]
@KweezyCode Carsten can say same thing on his end. We need more debugging. Can anyone user traceroute or something like this (its not my strong side)?

**Comment by C-Otto** [comment]
Here's how a `tcptraceroute 137.226.34.46 443` should look like:

```
[...]
 8  cr-fra2-be5.x-win.dfn.de (188.1.242.9)  4.198 ms  4.098 ms  4.119 ms
 9  kr-aah15-0.x-win.dfn.de (188.1.242.110)  7.579 ms  7.314 ms  7.277 ms
10  fw-xwin-1-vl106.noc.rwth-aachen.de (134.130.3.236)  7.287 ms  7.012 ms  7.026 ms
11  n7k-ww10-1-vl158.noc.rwth-aachen.de (134.130.3.243)  7.952 ms * *
12  * n7k-ww10-2-et1-1.noc.rwth-aachen.de (137.226.139.42)  8.083 ms *
13  * ftp.halifax.rwth-aachen.de (137.226.34.46) <syn,ack>  7.390 ms *
```

**Comment by C-Otto** [comment]
@Stikus I'm not involved with Jenkins, you might want to ask them to exclude our server (create a new issue in this repository).

**Comment by KweezyCode** [comment]
![Screenshot_20240610-123423](https://github.com/jenkins-infra/helpdesk/assets/36813920/f201f9c8-2054-4b2f-b403-76a83e4bca64)
![Screenshot_20240610-123639](https://github.com/jenkins-infra/helpdesk/assets/36813920/97d54888-2c60-4582-ad19-0bd5e6762957)

**Comment by KweezyCode** [comment]
connection refused means that connection to server is ok :/

**Comment by C-Otto** [comment]
`traceroute` does not accept the port (443) as an argument, either use `tcptraceroute` or drop the port. It looks like it is sending 443 byte packets instead, which is weird.

And: please don't post pictures of text.

**Comment by C-Otto** [comment]
Connection refused does not mean anything. It can be a firewall that cuts the connection - and I believe that's exactly what's happening here.

**Comment by C-Otto** [comment]
Oh, and one more thing - sorry for the comment spam. If "traceroute" succeeds, that just means that ICMP packages are routed correctly. That doesn't mean that TCP connections work. You might want to use `tcptraceroute` instead, for that reason.

**Comment by C-Otto** [comment]
If any of you is interested, we can do a live screensharing session where you can see the output of `tcpdump` for specific IPs. If you have access to both a Russian IP and some non-blocked non-Russian IP, we might be able to see the difference between ICMP packets (working fine) and TCP packets (blocked somewhere in Russia).

**Comment by KweezyCode** [comment]
![image](https://github.com/jenkins-infra/helpdesk/assets/36813920/66dcc092-865b-4513-9594-eb46a2be0500)

**Comment by C-Otto** [comment]
@KweezyCode is there a specific reason why you post pictures of text? That's just weird.

**Comment by KweezyCode** [comment]
> @KweezyCode is there a specific reason why you post pictures of text? That's just weird.

I'm just too lazy to copy the text, so I take a screenshot of the area and just paste. I can copy the text if it bothers you

**Comment by C-Otto** [comment]
Na, it's fine. I'll try to be lazy, too.

**Comment by dduportal** [comment]
Hi @KweezyCode @Stikus @ihatethecloud, we've finally enabled (a few minutes ago) the Hostico mirror in Romania.

Could you share with us the result of https://get.jenkins.io/windows-stable/2.452.1/jenkins.msi?mirrorlist to see how is the mirror redirector behaving in your areas?

It should shows Hostico with a higher priority than Aaechen University's mirror: could you confirm?

**Comment by dduportal** [comment]
Hi @w-e-g we could you check my message above please and share the result?

**Comment by KweezyCode** [comment]
works fine after changing mirror

**Comment by dduportal** [comment]
> works fine after changing mirror
> 
> UPD: still fails with some plugins, but downloads them much faster ![image](https://private-user-images.githubusercontent.com/36813920/338231475-3dcfd251-b94c-4181-8d4d-63f4887c1536.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgwMzEwMjgsIm5iZiI6MTcxODAzMDcyOCwicGF0aCI6Ii8zNjgxMzkyMC8zMzgyMzE0NzUtM2RjZmQyNTEtYjk0Yy00MTgxLThkNGQtNjNmNDg4N2MxNTM2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjEwVDE0NDUyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ4OWY5MzY2ZjU0YjQzMmE3NTE1ZDg0MTNmMjBmMzA0OWE2NzgxMjNhZDQ5YTE2NzZiZGQ1M2FhY2IxYjE5YmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.7DG4Xl9wop9aXHBzZtjWRtCCDGe48XZhlQGA9i2Vr2Y)
> 
> ```
> java.io.IOException: Failed to load: Pipeline: GitHub Groovy Libraries (pipeline-github-lib 61.v629f2cc41d83)
>  - Failed to load: Git plugin (git 5.2.2)
> 	at hudson.PluginWrapper.resolvePluginDependencies(PluginWrapper.java:992)
> 	at hudson.PluginManager.dynamicLoad(PluginManager.java:949)
> Caused: java.io.IOException: Failed to install pipeline-github-lib plugin
> 	at hudson.PluginManager.dynamicLoad(PluginManager.java:963)
> 	at hudson.model.UpdateCenter$InstallationJob._run(UpdateCenter.java:2249)
> Caused: java.io.IOException: Failed to dynamically deploy this plugin
> 	at hudson.model.UpdateCenter$InstallationJob._run(UpdateCenter.java:2253)
> 	at hudson.model.UpdateCenter$DownloadJob.run(UpdateCenter.java:1899)
> 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
> 	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
> 	at hudson.remoting.AtmostOneThreadExecutor$Worker.run(AtmostOneThreadExecutor.java:121)
> 	at java.base/java.lang.Thread.run(Unknown Source)
> 	```
> ```

Do you have the URL effectively used when failing? The snippet you show is only a partial of the whole logs so it's hard to diagnose.

**Comment by w-e-g** [comment]
@dduportal, just tested my script for mirroring plugins directory. Looks like connections are redirected to new mirror.

Btw, is new mirror have any limitations? Connections, address pool, etc. And which is the best way to sync Jenkins plugins? Rsync, http?

**Comment by dduportal** [comment]
> @dduportal, just tested my script for mirroring plugins directory. Looks like connections are redirected to new mirror.
> 

Hi @w-e-g , thanks for the confirmation. We'll close the issue again.

Many thanks @C-Otto for the support. Note that with the effective enablement of the new mirror in Romania, you should see a load decrease on your mirror.

> Btw, is new mirror have any limitations? Connections, address pool, etc.

@w-e-g We don't know: we are a sponsored project so we take what is given. You may want to contact Hostico for details on this.

> And which is the best way to sync Jenkins plugins? Rsync, http?

What do you mean by "sync Jenkins plugins" exactly? It looks like you are trying to build your own local mirror? If you are able to provide one in Russia, that would help you and other users as well: we need an HTTPS public server with around 500 to 750 Gb disk, and we also need an rsync or FTP server so we can scan the content of your mirror (to make the mirror redirector system working). Happy to give you details if you are interested to help the community

**Comment by mikhirev** [comment]
TCP tracerute breaks immediately after my ISP network

```
% sudo traceroute --tcp --port=443 ftp.halifax.rwth-aachen.de
traceroute to ftp.halifax.rwth-aachen.de (137.226.34.46), 30 hops max, 60 byte packets
 1  OpenWrt.home (192.168.1.1)  0.310 ms  0.480 ms *
 2  172.30.174.1 (172.30.174.1)  2.951 ms  2.976 ms  2.982 ms
 3  p2sr-e38sr.line-r.ru (213.108.208.98)  2.992 ms  3.003 ms  3.014 ms
 4  e38sr-g1.line-r.ru (213.108.208.169)  3.025 ms  3.045 ms  3.050 ms
 5  g1-e38sri.line-r.ru (213.108.208.166)  3.056 ms  3.188 ms  3.201 ms
 6  * * *
 7  * * *
 8  * * *
 9  * * *
10  * * *
11  * * *
12  * * *
13  * * *
14  * * *
15  * * *
16  * * *
17  * * *
18  * * *
19  * * *
20  * * *
21  * * *
22  * * *
23  * * *
24  * * *
25  * * *
26  * * *
27  * * *
28  * * *
29  * * *
30  * * *
```

This server is not present in the [static blocking list][1]. This means that it is blocked by TSPU that is a black box for ISPs and is directly controlled by Roskomnadzor. Usually when HTTPS sites are blocked, the TCP session is being reset, however in this case it seems that packets are simply dropped. So it is likely that the block is caused by the Tor node, not the website content.

I see only two solutions:

1. Do not redirect users from Russia to this mirror.
2. Serve the Jenkins mirror and the Tor node from different IP addresses.

[1]: https://raw.githubusercontent.com/zapret-info/z-i/master/dump.csv

**Comment by C-Otto** [comment]
Third and obvious option: have a nice chat with Roskomnadzor and kindly ask them to remove the block.

**Comment by dduportal** [comment]
- The problem is not related to the Jenkins infrastructure and neither to the Aachen University. As the root cause as been identified to be a Roskomnadzor block, please check with them
- As explained in https://github.com/jenkins-infra/helpdesk/issues/4128#issuecomment-2158518511, we provided a mirror close to Russia and at least 3 Russian users did confirm it work for them

=> We've done everything we could. If you still have issues, please help us by sponsoring the project and finding an organization, within the Russian network, which could provide us a mirror (see comments below) for Russian users.

**Comment by KweezyCode** [comment]
> Third and obvious option: have a nice chat with Roskomnadzor and kindly ask them to remove the block.

haha that's not how it works :D

**Comment by mikhirev** [comment]
> If you still have issues, please help us by sponsoring the project and finding an organization, within the Russian network, which could provide us a mirror (see comments below) for Russian users.

There's a mirror [at Yandex][1]. Is it unofficial? I can try to contact them if you tell me what they need to do to make it used by default for Russian users.

[1]: https://mirror.yandex.ru/mirrors/jenkins/plugins/

**Comment by w-e-g** [comment]
@mikhirev maybe I can help you. I have a contact of mirror.yandex.ru engineer. But it has been a very long time ago and I'm not sure that he's work on this today.

**Comment by dduportal** [comment]
@mikhirev @w-e-g interesting! I confirm it is an unofficial mirror: but if they can provide us (Jenkins infra team - contact email at jenkins-infra-team <at> googlegroups.com) an Rsync or FTP (so we can scan files and setup the redirector system) then we could absolutely add it as another mirror!

**Comment by lemeurherve** [comment]
@mikhirev @w-e-g that would be great!

For adding a mirror we need:
- an URL where the mirror will be available
- a rsync address and/or a FTP address
- an administrative email address

Current size of all mirrored files is around 530Gio.

For the initial sync, we suggest using OSUOSL mirror:

https://ftp-nyc.osuosl.org/pub/jenkins
rsync://ftp-nyc.osuosl.org/jenkins
ftp://ftp-nyc.osuosl.org/pub/jenkins

**Comment by dduportal** [comment]
Hey @mikhirev @w-e-g and others: I've opened https://github.com/jenkins-infra/helpdesk/issues/4147 to work specifically on mirros for Jenkins Russian users if you are interested to help or at least being informed!

**Comment by timja** [comment]
FYI Russia mirror completed in https://github.com/jenkins-infra/helpdesk/issues/4147#issuecomment-2186675301
